src/main/java/org/elasticsearch/indices/recovery/ShardRecoveryHandler.java

```diff
@@ -345,10 +350,49 @@ public void run() throws InterruptedException {
                     // Once the files have been renamed, any other files that are not
                     // related to this recovery (out of date segments, for example)
                     // are deleted
-                    transportService.submitRequest(request.targetNode(), RecoveryTarget.Actions.CLEAN_FILES,
-                            new RecoveryCleanFilesRequest(request.recoveryId(), shard.shardId(), recoverySourceMetadata),
-                            TransportRequestOptions.options().withTimeout(recoverySettings.internalActionTimeout()),
-                            EmptyTransportResponseHandler.INSTANCE_SAME).txGet();
+                    try {
+                        transportService.submitRequest(request.targetNode(), RecoveryTarget.Actions.CLEAN_FILES,
+                                new RecoveryCleanFilesRequest(request.recoveryId(), shard.shardId(), recoverySourceMetadata),
+                                TransportRequestOptions.options().withTimeout(recoverySettings.internalActionTimeout()),
+                                EmptyTransportResponseHandler.INSTANCE_SAME).txGet();
+                    } catch (RemoteTransportException e) {
+                        IOException corruptIndexException;
+                        // we realized that after the index was copied and we wanted to finalize the recovery
+                        // the index was corrupted:
+                        //   - maybe due to a broken segments file on an empty index (transferred with no checksum)
```

Comment:
Can you explain this case to me? I am just curious, because all commits we need to read (4.0+) have a CRC32 checksum as the last 8 bytes. Its just that since lucene 4.8 we also added the rest of the footer (FOOTER_MAGIC) to be consistent with all other lucene files in structure. So if we want, we can always at least just check that the last 8 bytes are the CRC32 checksum for any segments_N (just don't look for a footer or call checkFooter).